packages/fxa-settings/src/components/UnitRowRecoveryKey/index.tsx

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { useBooleanState } from 'fxa-react/lib/hooks';
+import AlertBar from '../AlertBar';
+import Modal from '../Modal';
+import UnitRow from '../UnitRow';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import { useAccount } from '../../models';
+
+export const DELETE_RECOVERY_KEY_MUTATION = gql`
+  mutation deleteRecoveryKey($input: DeleteRecoveryKeyInput!) {
+    deleteRecoveryKey(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const UnitRowRecoveryKey = () => {
+  const { recoveryKey } = useAccount();
+  const [modalRevealed, revealModal, hideModal] = useBooleanState();
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [errorText, setErrorText] = useState<string>();
+  const onError = (e: Error) => {
+    setErrorText(e.message);
```

Comment:
idk yet which apollo errors we should be sending to sentry, but yeah we should set a convention. I don't _think_ any of the errors we get at this level would be actionable for us, but maybe we still want a count 🤷 

as for graphqlerrors specifically I think we'll want to handle those on a case by case basis. in this case an invalid code error will be quite common

Reply:
Makes sense. I think I'll update #6249 to include something around setting our convention then, it may be show the specific message coming from a GQL error and otherwise, show a generic "there was a problem, please try again" (or whatever Meridel wants to show) error message.

I'll ask around about what should report to Sentry and and add it to that ticket as well